### PR TITLE
unify auth and permission configuration

### PR DESCRIPTION
**Pull request by jwietelmann**
* Upgrade `dex` HelmRelease
* Create `user-permission` HelmRelease
* Remove ClusterRoles and ClusterRoleBindings from `dex.yaml`
* Modify `values.yaml` format to:
  * Provide `roleDefinitions` for the `user-permissions` chart
  * Provide an `auth` key for defining Dex connectors and associated group/team permissions at the same time.
* Create `docs/README.md` for a more thorough configuration guide.
* Remove config params table from top-level `README.md`.
* Edit top-level `README.md` to refer to `docs/README.md`.
* Other little edits to `README.md`

### CODE REVIEW

**Review by bit-herder — COMMENTED** [review]
were you able to test this against a current cluster yet?

**Comment by jwietelmann** [comment]
@bit-herder I did a lot of install/uninstall/test/repeat on my test cluster. I promise I'll do it again before merging.

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+Configures the [`flux`](https://github.com/fluxcd/flux/tree/master/chart/flux) chart for enabling GitOps. This will allow you to manage cluster resources simply by committing YAML files to a git repository.
+
+| name | type | description |
+| `----` | ---- | ----------- |
+| `flux.create` | boolean (optional; default `true`) | Whether to create this HelmRelease |
+| `flux.namespace` | string (optional) | The namespace for the HelmRelease |
+| `flux.releaseName` | string (optional) | The release name for the HelmRelease |
+| `flux.git.repo` |  |  |
+| `flux.git.branch` |  |  |
+| `flux.auth.key` |  |  |
```
Comment (grossvogel) [inline docs/README.md:152]: I think a few of these options that are blank could use some more explanation (here and in `harbor` for example). Could definitely be a follow-on of course

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+
+| name | type | description |
+| `----` | ---- | ----------- |
+| `gangway.create` | boolean (optional; default `true`) | Whether to create this HelmRelease |
+| `gangway.namespace` | string (optional) | The namespace for the HelmRelease |
+| `gangway.releaseName` | string (optional) | The release name for the HelmRelease |
+
+## harbor
+
+Configures the [`harbor`](https://github.com/goharbor/harbor-helm) chart.
```
Comment (grossvogel) [inline docs/README.md:166]: "... to host an in-cluster docker registry."

**Review by iamjoshfrank — APPROVED** [review]
Reviewed the `README.md` and `docs/README.md` changes. These updates look helpful and I like the approach of starting to break the documentation out into specific files.

Can't comment on the `.yaml` file updates, other than to ask a few questions related to documentation (I did a bit of Google searching to try to answer these two questions on my own, but came up short in the time-box I allowed):

1. Would it be helpful to have more of these comments / instructions in the actual `.yaml` files?
2. Does YAML have something like Elixir's `@moduledoc`, `@doc` and a tool like ExDoc that can generate documentation?

Even if the answers to 1. and 2. are "yes," I don't think that should block us on shipping these updates and opening up an Issue or Issues to plan out and prioritize 1. and 2.

**Review by grossvogel — APPROVED** [review]
With the caveat that I'm not a helm expert (I followed it I just don't know it well enough to find bugs or critique style), I couldn't find anything about this that I don't love, except a few doc nits.

The doc work addresses basically everything important I had to say in #10 already

**Comment by jwietelmann** [comment]
@iamjoshfrank 1) yes 2) no

I agree but I think a bunch of this is changing shortly, and my README updates here are still missing a few things (that were already missing) so I'm holding on it until some of the other open issues are resolved.

**Comment by iamjoshfrank** [comment]
Word. 

I'm totally on-board with incremental improvement and gradual re-organization/optimization. Especially if we continue to split out individual Issues to outline what we think needs improved.

Thanks for those answers!

**Comment by jwietelmann** [comment]
it's working! merging...

**Pull request merged**